improver/calibration/ensemble_calibration.py

```diff
@@ -562,42 +563,57 @@ def __repr__(self):
+        """Create a cubelist for storing the coefficients computed using EMOS.
 
         .. See the documentation for examples of these cubes.
         .. include:: extended_documentation/calibration/
            ensemble_calibration/create_coefficients_cube.rst
 
         Args:
-            optimised_coeffs (list):
+            optimised_coeffs (numpy.ndarray):
                 List of optimised coefficients.
                 Order of coefficients is [gamma, delta, alpha, beta].
             historic_forecast (iris.cube.Cube):
                 The cube containing the historic forecast.
 
         Returns:
-            iris.cube.Cube:
-                Cube constructed using the coefficients provided and using
-                metadata from the historic_forecast cube.  The cube contains
-                a coefficient_index dimension coordinate where the points
-                of the coordinate are integer values and a
-                coefficient_name auxiliary coordinate where the points of
-                the coordinate are e.g. gamma, delta, alpha, beta.
+            iris.cube.CubeList:
+                CubeList constructed using the coefficients provided and using
+                metadata from the historic_forecast cube. Each cube within the
+                cubelist is for a separate EMOS coefficient e.g. gamma, delta,
+                alpha, beta.
 
         Raises:
             ValueError: If the number of coefficients in the optimised_coeffs
                 does not match the expected number.
         """
+        coeff_names = self.coeff_names
         if self.predictor.lower() == "realizations":
-            realization_coeffs = []
-            for realization in historic_forecast.coord("realization").points:
-                realization_coeffs.append(
-                    "{}{}".format(self.coeff_names[-1], np.int32(realization))
+            realizations_coeff_names = coeff_names[:-1] + [coeff_names[-1]] * len(
+                historic_forecast.coord("realization").points
+            )
+            beta_values = np.array([], dtype=np.float32)
+            for optimised_coeff, coeff_name in zip(
+                optimised_coeffs, realizations_coeff_names
+            ):
+                if coeff_name == "beta":
+                    beta_values = np.append(beta_values, optimised_coeff)
+            if len(beta_values) != len(historic_forecast.coord("realization").points):
+                msg = (
+                    "The number of beta coefficients in {} must equal the "
+                    "number of realizations {}, when the predictor is "
+                    "'realizations'.".format(
+                        optimised_coeffs[-len(beta_values) :],
+                        historic_forecast.coord("realization").points,
+                    )
                 )
-            coeff_names = self.coeff_names[:-1] + realization_coeffs
-        else:
-            coeff_names = self.coeff_names
+                raise ValueError(msg)
```

Comment:
I've simplified this check and separated this out from the `create_coefficients_cubelist` method into a separate method.

Reply:
Additionally following these and other changes to the `beta` coefficient, I've straightened out the ordering of the coefficients used throughout the code to be alpha, beta, gamma, delta, rather than the previous gamma, delta, alpha, beta.